Source/utils/string_variant.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <string>
+#include "utils/stdcompat/string_view.hpp"
+
+namespace devilution {
+
+class StringVariant
+{
+	std::string str;
+	string_view view;
+
+public:
+	StringVariant()
+	{
+	}
+
+	explicit StringVariant(string_view view)
```

Comment:
Rather than these constructors, I think it maybe clearer to have factory methods:

```c++
StringVariant::Owned(...);
StringVariant::View(...);
```

Otherwise it is not obvious at call sites which one is being constructed

Reply:
That's even more tedious than `std::variant<string, string_view>`. At least with `variant`, you only have to worry about `const char *` and `char[]`; the others you can assign directly.

The long term plan is to get rid of using char for strings (and these constructors along with them).

Reply:
Tedious is fine, code is read more than it is written!

Reply:
And with factory methods like this, you wouldn't need `char *` overloads at all, just a `string_view` one.

Reply:
It's tedious to read, as well.

Reply:
Also more error prone. It's now on the programmer to find the right type of StringVariant. Made an unowned one from a std::string? The game may display garbage.